### DISCUSSION TITLE
Adicionar uma opção de CONTRATOS

### DISCUSSION THREAD

**Discussion by Deniizynhu** [RamonSilva20/mapos#1583] [Ideas]
Gostaria de parabenizar a equipe de desenvolvimento, pois o sistema é muito bom e está superando as nossas expectativas.
Gostaria de solicitar que, se possível, fosse criado uma guia para a adição de contratos.
Estou abrindo uma empresa de suporte e soluções, e estou usando o sistema MAP-OS de forma online, e gostaria de que fosse adicionado a opção de CONTRATOS, gerar contrato, adicionar contrato na O.S e/ou em um determinado cliente.

Isso é possível?

Aguardo retorno!

**Comment by cleytonasa** [comment]
Gostei da sugestão!!

**Comment by SuperPorte** [comment]
Se vir com uma opção de tabela de preços diferenciados, melhor ainda :D

**Comment by klessioand94** [comment]
eu conseguir fazer a pagina para gerar o contrato, o problema maior é o layout do contrato, fiz também pagina de orçamento, e entre outras coisas.

**Reply by Deniizynhu** [reply]
Consegue me mandar um print ou alguma amostra?
Talvez possa te ajudar na questão do Layout.
Mas se torna algo muito interessante.
E eu esqueci também de comentar que seria muito importante um campo na O.S para relacionar todos os custos da O.S.
Tipo como atendemos no local solicitado pelo cliente cobramos a condução (ônibus) ou o combustível. Mas isso não conseguimos mensurar na O.S caso não seja lançado como SERVIÇO.

**Comment by klessioand94** [comment]
Amanha te envio, mas resumindo, eu criei novas funções "imprimir"...

Em qua., 12 de jan. de 2022 às 21:49, Deniizynhu ***@***.***>
escreveu:

**Reply by Deniizynhu** [reply]
Tudo bem, fico no aguardo.
Me manda também qualquer dúvida que você tiver, idéias.
Assim eu consigo te ajudar de uma forma mais completa.
Sou gestor de Tecnologias, quem sabe consigo agregar algo a mais no excelente sistema que foi criado!

**Reply by klessioand94** [reply]
Pagina dos clientes onde tem o botão para imprimir o contrato.

![CONTRATO 2](https://user-images.githubusercontent.com/96914165/149502123-c303f20d-e7ed-4c4b-99ed-935c6bb8af42.png)

Função criada para ler a pagina que imprime o contrato.

![CONTRATO 1](https://user-images.githubusercontent.com/96914165/149502236-772fb030-1748-4b6c-98f1-ca5c1bfbf33a.png)


O contrato é impresso de acordo com as informações inseridas na database, eu fiz modificações na pagina de cadastro do cliente para atender minhas necessidades. Estou rodando o mapos no ubuntu.

**Reply by klessioand94** [reply]
A questão do layout do contrato é porque eu tive que criar o html através do word, também criei pagina com orçamento personalizado.

**Reply by Deniizynhu** [reply]
Desculpa, sou muito leigo em programação.
Onde que é realizadas essas alterações?
Não entendi como fazer. Sou muito horrível com programação.

**Reply by klessioand94** [reply]
Em controllers.

**Reply by borgesthiago** [reply]
No symfony tem um bundle (não lembro nome) que já usei que faz o seguinte:

- Precisa criar 1 docx e inclui nele tag "${razao}"... já dentro do modelo do contrato;
- Fiz upload ficando como modelo e dps era só clicar p imprimir que gerava o contrato completo com os dados.
Acho que vale a pesquisa.
